orttraining/orttraining/python/orttraining_pybind_state.cc

```diff
@@ -760,6 +772,42 @@ void addObjectMethodsForTraining(py::module& m, ExecutionProviderRegistrationFn
         return ortmodule_graph_builder->GetGraphInfo();
       });
 
+  // Provide a convenient and well-documented way to make a gradient graph.
+  // It's possible to get the gradient graph through ORTModule by leveraging some "private" fields and not-so-well-documented APIs, so we provide this explicit and tested way to get the gradient graph.
+  py::class_<PyGradientGraphBuilder> gradient_graph_builder(m, "GradientGraphBuilder", R"pbdoc(A utility for making a gradient graph that can be used to help train a model.)pbdoc");
+  // Set up methods to match the C++ `GradientGraphBuilder` interface.
+  gradient_graph_builder.def(py::init([](
```

Comment:
Sorry I'm not actually that familiar with this code base, is there a specific type that you had in mind for "onnx graph"? That would be great if the developer didn't have to `torch.onnx.export(...` to a file.

As for the idea itself, I'm not sure how useful that will be since this is to expose a convenient way to use `GradientGraphBuilder` to Python. `GradientGraphBuilder` does take in `onnxruntime::Graph`.

Reply:
It is sometime useful to avoid writing onnx file on disk and it is faster for unit tests.